ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/session-timeout/session-timeout.tsx

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+
+import * as React from 'react'
+import styled from '../../styles/styled-components'
+const sessionTimeoutModel = require('../../../component/singletons/session-timeout')
+import { Button, buttonTypeEnum } from '../button'
+
+const SessionTimeoutRoot = styled.div`
+  height: 100%;
+  width: 100%;
+  display: block;
+  overflow: hidden;
+`
+const Message = styled.div`
+  max-height: 'calc(100% - 2.25rem)';
+  height: auto;
+  text-align: center;
+  padding: ${props => props.theme.mediumSpacing};
+`
+
+type State = {
+  timeLeft: number
+}
+const renewSession = () => {
+  sessionTimeoutModel.renew()
+}
+
+class SessionTimeout extends React.Component<{}, State> {
+  constructor(props: any) {
+    super(props)
+    this.setState({ timeLeft: sessionTimeoutModel.getIdleSeconds() })
+  }
+  render() {
+    setTimeout(
```

Comment:
✏️ I think it would be better to have a listener for the backbone model and update the state when that changes. That involves importing WithBackboneProps as done [here](https://github.com/codice/ddf/blob/15f20494516fa01613593c0b4055c99bcf020a8e/ui/packages/catalog-ui-search/src/main/webapp/react-component/container/navigation-container/navigation-container.tsx#L14) and passing that in as props, creating a listener on the backbone model as done [here](https://github.com/codice/ddf/blob/15f20494516fa01613593c0b4055c99bcf020a8e/ui/packages/catalog-ui-search/src/main/webapp/react-component/container/navigation-container/navigation-container.tsx#L74) and exporting the module with `WithListenTo` as done [here](https://github.com/codice/ddf/blob/15f20494516fa01613593c0b4055c99bcf020a8e/ui/packages/catalog-ui-search/src/main/webapp/react-component/container/navigation-container/navigation-container.tsx#L113)